cdap-app-fabric/src/main/java/co/cask/cdap/app/stream/DefaultStreamWriter.java

```diff
@@ -163,6 +173,9 @@ public StreamBatchWriter createBatchWriter(String stream, String contentType) th
     connection.setReadTimeout(15000);
     connection.setConnectTimeout(15000);
     connection.setRequestProperty(HttpHeaders.CONTENT_TYPE, contentType);
+    if (kerberosEnabled) {
```

Comment:
how come we didn't need this till now? This seems unrelated to authorization, right?